cmd/earthly/main.go

```diff
@@ -2801,6 +2802,45 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
 			return errors.Wrapf(err, "parse target name %s", targetName)
 		}
 	}
+
+	cleanCollection := cleanup.NewCollection()
+	defer cleanCollection.Close()
+
+	cc, err := cloud.NewClient(app.apiServer, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	// Default upload logs, unless explicitly configured
+	if !app.cfg.Global.DisableLogSharing {
+		_, _, _, err := cc.WhoAmI()
+		isLoggedIn := err == nil
```

Comment:
We should perhaps detect that this error is actually about not being logged in, and not 500 / 404 or other such cases.

Reply:
I suppose we could append a warning if its anything other than a 401 next to the advertising message?

Reply:
I tried my hand at this. If its unauthorized, then we advertise. Otherwise we say something else.

Reply:
Did you mean to make a code change for this as well?

BTW, a `// TODO` is also fine if we want to just get this out

Reply:
The code change is lower in the file. It seemed like we would want to treat any error here as if we weren't logged in, and avoid attaching a logger because later upload would likely fail as well.

So, I added a conditional further down that branches on what kind of error we got. If it is `ErrUnauthorized`, we print the invitation message, otherwise, we print an error saying we could not upload the log.